src/core/geometry/qgsabstractgeometry.h

```diff
@@ -350,6 +357,29 @@ class CORE_EXPORT QgsAbstractGeometry
         \returns the converted geometry. Caller takes ownership*/
     virtual QgsAbstractGeometry *toCurveType() const SIP_FACTORY;
 
+    /** Makes a new geometry with all the points or vertices snapped to the closest point of the grid.
+     * It transfers ownership to the callee.
+     * If it couldn't make the gridified geometry it returns nullptr.
+     * It may generate an invalid geomety (in some corner cases).
+     * It can also be thought as rounding the edges and it may be useful for removing errors.
+     * If the geometry is curved, it will be segmentized before gridifying it.
+     * Example:
+     * \code
+     * geometry->asGridified(1, 1);
+     * \endcode
+     * In this case we use a 2D grid of 1x1 to gridify.
+     * In this case, it can be thought like rounding the x and y of all the points/vertices to full units (remove all decimals).
+     * \param hSpacing Horitzontal spacing of the grid (x axis). 0 to disable.
+     * \param vSpacing Vertical spacing of the grid (y axis). 0 to disable.
+     * \param dSpacing Depth spacing of the grid (z axis). 0 (default) to disable.
+     * \param mSpacing Custom dimension spacing of the grid (m axis). 0 (default) to disable.
+     * \param tolerance In case of segmentation, the tolerance to use (passed to segmentize as is).
```

Comment:
The idea was that if you have a `QgsAbstractGeometry` pointer, you can control the tolerance without casting (not all geometries can be segmentized).
Segmentize always makes a copy, even if you don't actually segmentize. This prevents it; only using new memory when required.
It also happens to prevent other people to writing loops for the `QgsCollection` (and subclasses) to segmentize with the desired tolerance.